MDANSE/Src/MDANSE/Framework/Formats/HDFFormat.py

```diff
@@ -28,6 +29,103 @@
     from MDANSE.Framework.OutputVariables.IOutputVariable import IOutputVariable
 
 
+json_decoder = json.decoder.JSONDecoder()
+
+
+def check_metadata(hdf5_file: h5py.File) -> dict[str, str]:
+    """Extract metadata from an MDANSE HDF5 file.
+
+    Parameters
+    ----------
+    hdf5_file : h5py.File
+        MDANSE output file, .mda or .mdt
+
+    Returns
+    -------
+    dict[str, str]
+        dictionary of saved input information used to create the file
+
+    """
+    meta_dict = {}
+
+    def put_into_dict(name: str, obj: bytes):
+        """Put an entry from an HDF5 dataset into a dictionary, as string.
+
+        This helper function is used together with the visititems method
+        of HDF5 datasets, provided by h5py. It will be called for each
+        dataset in the 'metadata' group, and it will try to convert
+        the contents of that dataset to string, which will then be stored
+        in the meta_dict dictionary.
+
+        Parameters
+        ----------
+        name : str
+            name (key) of the dataset from an HDF5 group
+        obj : bytes
+            contents of the dataset (text stored as 'bytes')
+        """
+        try:
+            string = obj[:][0]
+        except TypeError:
+            try:
+                string = obj[0]
+            except TypeError:
+                return
+        try:
+            string = string.decode()
+        except KeyError:
+            LOG.debug(f"Decode failed for {name}: {obj}")
+            meta_dict[name] = str(obj)
+        else:
+            try:
+                meta_dict[name] = json_decoder.decode(string)
+            except ValueError:
+                meta_dict[name] = string
+
+    try:
+        meta = hdf5_file["metadata"]
+    except KeyError:
+        return
+    else:
+        meta.visititems(put_into_dict)
+
+    meta_dict["<b>file header</b>"] = "\n" + hdf5_file.attrs.get("header", "no header")
+
+    return meta_dict
+
+
+def write_metadata(job: "IJob", output_file: h5py.File):
```

Comment:
This seems like it might want to be a method on `IJob`, though I'm not 100% sure since it makes sense here too...